AEPNotificationContent/Sources/Model/CarouselPayload.swift

```diff
@@ -0,0 +1,81 @@
+/*
+ Copyright 2024 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+import UIKit
+import UserNotifications
+
+class CarouselPayload: Payload {
+    // MARK: - Properties
+
+    /// The mode of the carousel.
+    let carouselMode: CarouselMode
+
+    /// The layout of the carousel.
+    let carouselLayout: CarouselLayout
+
+    /// The items in the carousel.
+    var carouselItems: [CarouselItem] = []
+
+    /// Initializes a `CarouselPayload` instance from a `UNNotificationContent`.
+    /// Initialization fails if the mandatory properties required for CarouselTemplate are unavailable.
+    /// - Parameter notificationContent: The content of the notification.
+    required init?(from notificationContent: UNNotificationContent) {
```

Comment:
nit: remove `from` as the label in this initializer

Reply:
Unfortunately. I will not be able to do that.

The parent class of CarouselPayload -> Payload has a non-failable initializer with the definition
```
init(notificationContent: UNNotificationContent) 
```
 I cannot override the above because this initializer is a failable one